ocean/src/social.rs

```diff
@@ -31,14 +57,33 @@ impl SocialNetwork {
     * Result enum
     */
     pub fn remove_friend(&mut self, crab_id: String, friend_id: String) -> Result<(), String> {
-        unimplemented!();
+        match self.friendships.get_mut(&crab_id) {
+            Some(friends) => {
+                if friends.contains(&friend_id) {
+                    friends.retain(|f| f != &friend_id);
+                    Ok(())  // Friend was found and removed
+                } else {
+                    Err("Friend not found in list".to_string())  // Friend was not in the list
+                }
+            },
+            None => Err("Crab ID does not exist or has no friends".to_string())  // Crab ID does not exist or has no friends
+        }
     }
 
     /*
     Returns True if crab2 is a friend of crab1, else False.
     */
-    pub fn is_friend(&self, crab_id: String, friend_id: String) -> bool {
-        unimplemented!();
+    pub fn is_friend(&self, crab1_id: String, crab2_id: String) -> bool {
+        match self.friendships.get(&crab1_id) {
+            Some(friends) => {
+                if friends.contains(&crab2_id) {
+                    return true
+                } else {
+                    return false
+                }
```

Comment:
We  could move `return false` outside the `else{}` here and get rid of the block entirely